client/consensus/aura/src/lib.rs

```diff
@@ -264,11 +272,30 @@ impl<B, C, E, I, P, Error, SO> sc_consensus_slots::SimpleSlotWorker<B> for AuraW
 		StorageChanges<sp_api::TransactionFor<C, B>, B>,
 		Self::Claim,
 		Self::EpochData,
-	) -> sp_consensus::BlockImportParams<B, sp_api::TransactionFor<C, B>> + Send> {
-		Box::new(|header, header_hash, body, storage_changes, pair, _epoch| {
+	) -> Result<sp_consensus::BlockImportParams<B, sp_api::TransactionFor<C, B>>, sp_consensus::Error> + Send + 'static> {
+		let keystore = self.keystore.clone();
+		Box::new(move |header, header_hash, body, storage_changes, pair, _epoch| {
 			// sign the pre-sealed hash of the block and then
 			// add it to a digest item.
-			let signature = pair.sign(header_hash.as_ref());
+			let public = pair.public().to_raw_vec();
+			let public_type_pair = CryptoTypePublicPair(
```

Comment:
This is really verbose, why does the public key not directly implements `to_public_pair()`? Or something similar?

Reply:
I looked around and i think you're right. We could introduce a `to_public_crypto_pair` as you suggested into the `Public` trait and use that one.